CONTRIBUTING.md

```diff
@@ -120,213 +122,73 @@ the team support them (though the plugin may work, and will usually install).
 *IDEA releases 2016 and later require JDK 8.  That build environment has
 been successfully used for this plugin, targeting Java6 for builds prior to 2016.x.*
 
-#### Ant Builds
+#### Gradle Builds
 
 As we noted in the [README](./README.md) file, you can build and test the 
-plugin without ever installing IDEA Ultimate.  The ```make``` command does
-this for you -- every time you build.  That is rather inefficient and can 
-be time consuming, particularly when we, as developers, generally already 
-have the software installed.
+plugin without ever installing IDEA Ultimate.  The ```gradlew``` command will 
+do everything for you. The first time you run the command for a given IDEA version 
+it will download  IDEA and all dependencies for that version. Any later executions 
+of this command  should  execute much faster as it will reused the already downloaded files.
```

Comment:
`Clean` does not remove downloaded versions of intelliJ
(only the `removeIdeaCache` task will delete previous versions)
as you can see from the image below i still have older versions even though i  target 2018.
![image of multiple versions of IDEA](https://www.m0rkeulv.net/images/haxe/multipleVersions2.png)

The plugin downloads IntelliJ from Jetbrains repository and currently they have releases all the way back to build 139.1.20 (released back in 2014) so i don't think this will be an issue any time soon.

Jetbrains repository  : https://www.jetbrains.com/intellij-repository/releases

I have not tested this  but i think unziping  a release to the expected folder should work fine, if that's not the case then a few changes in the build script would fix that, but as mentioned above i don't think  this will be needed anytime soon.

Reply:
First, that repository only contains the Windows releases.  You can't install to Mac or Linux from there (they don't have the Java binaries, for instance).  Second, downloads from there are soooooo sloooooooowwwwwww.  If we actually used it for CI, it would take an hour to do a build/run on Travis, instead of 5 minutes.

I just want to be sure that when a customer tells me to test issues using 2018.3.3, I can do that and not have a "clean" build remove that version.

Reply:
To me these files looks like they are all-in-one packages, they work just fine on both my mac and my windows computer. 
![image-of-mac-linux-windows-folders](https://m0rkeulv.net/images/haxe/platform-independent.jpg)
The Jetbrains JRE is not included by default, from what I can tell it will by default use the same JRE as the one your current IntelliJ instance is using. If you need to use a specific Jetbrains JRE you can provide it in the `runIde` configuration 
```
runIde {
    jbreVersion 'jbrex8u152b1248.6'
}
```
> (Note that this will be downloaded and stored in Gradles cache and not the project directory)

I have tried to figure out what the main difference between the official openJDK and jetbrains version is, and as far as I can tell it’s just cosmetic stuff like HiDPI support, better font rendering and those kind of things, it should as far as I can tell not affect any plugin logic


When it comes to the speed of the repo, this does not seem to be an issue just look at the Travis build for this branch, all builds completed in approx. 5-6 minutes. From what I can tell files in this repo is also available on a CDN but you might have to manipulate the URL to fetch from the CDN servers (it looks like that’s what the plugin does)  

just add `/~urlswitch/`  after ` jetbrains.com`  in the file URL and you get decent download speed.

Ex.
https://www.jetbrains.com/~urlswitch/intellij-repository/releases/com/jetbrains/intellij/idea/ideaIC/181.4445.20/ideaIC-181.4445.20.zip

i get 25-30mb/s  when i use these urls :)
![image-of-30mbs-donwload-speed](https://m0rkeulv.net/images/haxe/repo-speed.png)

Reply:
`/~urlswitch/` is something that I did not know about.  Since the speed seems decent, then we'll go with it.

I don't remember the packages being multi-target out there.  I had even asked a question of why only Windows targets were out in the repository.  The JetBrains folk's anwer was, "Why do you need other targets for CI?"  That was back in the 14.1 days, so things might have changed since then.  Or... I just missed that they were really multi-target.